doc/ref_arch/openstack/chapters/chapter04.md

```diff
@@ -1,82 +1,209 @@
 
 
 <a name="4.4"></a>
-## 4.4 Cloud Controller Services
-- Core Components (keystone, heat, nova, neutron, glance)
-- Identity Management
+## 4.4 Compute Node Services
 
+In Section 3.5 the high-level set of infrastructure components needed to run VMs was presented.  This section is primarily about scheduling the VM to be created onto a particular physical server. This section delves into the compute host selection for these VMs, the resource pools, hardware and software considerations.
 
-<a name="4.5"></a>
-## 4.5 Cloud Workload Services.
+<p align="center"><img src="../figures/Figure_4_3_Select_host_instance_launch.png" alt="Selection of a compute host to launch an instance"></br>Figure 4-3. Selection of a compute host to launch an instance.</p>
+
+When a request for an instance creation is made, the requested features and capabilities of the instance is used to determine the host on which the instance should be launched (Figure 4-3).  The nova scheduler service matches requested features and capabilities of the instance and the capabilities/configuration of the hosts (hardware and software).  The nova-scheduler can be configured to use the host aggregates (and availability zones) in selecting the hosts when an instance create request with these capabilities/configurations is requested. The Common Telco NFVI has only a handful of standard profiles and hence the number of potential target host servers (Figure 4-3) may be large if only the host-aggregates filter is utilized and, thus, other filters should also be selected; affinity/non-affinity, custom key-value pairs, etc. 
+
+A flavor may, in addition to the base flavor properties, include additional properties specified as key-value pairs in the extra specifications section of the flavour definition.  These additional properties can specify advanced configurations information or desired hardware characteristics. Examples include SSD drives, hardware accelerators, or a key-value pair used as meta data to associate the host with a tenant or host aggregate; both the host aggregate and the hosts that are to be assigned to the host aggregate would be assigned the same key-value pair, for example, <OVS-DPDK, True>. 
+
+<a name="4.4.1"></a>
+### 4.4.1. Hardware Considerations
 
-- Basic Profile
-  - Software components
-    - > for example: vSwitch.
-  - Hardware Platform
-    - > for example: single socket platform. 2x10G NIC, etc.
+-	Multiple pools of hardware resources where each resource pool caters for workloads of a specific profile (for example, network intensive). Leads to efficient use of the hardware as the server resources are specific to the flavour.  If not properly sized or when demand changes can lead to oversupply/starvation scenarios; reconfiguration may not be possible because of the underlying hardware or inability to vacate servers for reconfiguration to support another flavor type. The specifications for this type of resource pooling is specified in 4.5.2.
```

Comment:
This is too generic. This chapter should describe if there are different hardware profiles allowed or there is a single HW profile supported for this RA. 
Different HW profiles might be needed to support the following optional features: 

- e.nfvi.per.cap.003 IPSec Acceleration if optional in case of N Instances
- e.nfvi.per.cap.004 Crypto Acceleration is optional in case of N Instances
- e.nfvi.per.cap.005 Transcoding Acceleration is optional in case of C Instances
- e.nfvi.per.cap.006 Programmable Acceleration is optional for C Instances

Reply:
Agree with you Gergely. we decided to approve it and add details in separate PR.

Reply:
IMHO, there are choices that can be made by Telco providers. It is not necessary that the RA only specify whether we support a single host profile or multiple host profiles. If we were to distinguish RAs at that level of granularity we will have a proliferation of RAs for the same technology (e.g., OpenStack). It may be better, in my opinion, to provide the choices, rationale, pros/cons for that choice. 

This also holds for whether we want to containerise OpenStack services or not. Whether to use Cinder or Swift for image storage, etc.

Reply:
I think these are the things exactly what a RA should describe. 
Lets continue the discussion in #339

Reply:
@CsatariGergely Agree that if the optional services were supported you will need a different host profile. The discussion is about choices that operators can make -- if an operator were building a "public" type cloud (sufficient workload volume for each instance type) multiple profiles may well be necessary. However, for a pure telco cloud,  for both resource and operational efficiency a single profile may be advantageous allowing "any host any workload."

Reply:
I think this chapter should state if different host profiles are allowed and should describe these different host profiles.